gobblin-aws/src/main/java/gobblin/aws/GobblinAWSClusterMaster.java

```diff
@@ -0,0 +1,160 @@
+package gobblin.aws;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.apache.hadoop.fs.Path;
+import org.apache.helix.HelixManager;
+import org.apache.helix.NotificationContext;
+import org.apache.helix.messaging.handling.HelixTaskResult;
+import org.apache.helix.messaging.handling.MessageHandler;
+import org.apache.helix.messaging.handling.MessageHandlerFactory;
+import org.apache.helix.model.Message;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Optional;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+
+import gobblin.annotation.Alpha;
+import gobblin.cluster.GobblinClusterConfigurationKeys;
+import gobblin.cluster.GobblinClusterManager;
+import gobblin.cluster.GobblinHelixJobScheduler;
+import gobblin.cluster.HelixMessageSubTypes;
+import gobblin.cluster.JobConfigurationManager;
+
+
+/**
+ * The AWS Cluster master class for Gobblin.
+ *
+ * <p>
+ *   This class makes use of super class {@link GobblinClusterManager} to run:
+ *   1. {@link GobblinHelixJobScheduler} for scheduling and running Gobblin jobs.
+ *   2. {@link HelixManager} to work with Helix and act as Helix controller.
+ *   3. {@link JobConfigurationManager} to discover new job configurations and updates to
+ *   existing job configurations.
+ *
+ *   More AWS specific services can be added in future to this class that are required to be
+ *   run on Gobblin cluster master.
+ * </p>
+ *
+ * <p>
+ *   Note: Shutdown initiated by {@link GobblinAWSClusterLauncher} via a Helix message of subtype
+ *   {@link HelixMessageSubTypes#APPLICATION_MASTER_SHUTDOWN} is handled by super class {@link GobblinClusterManager}
+ * </p>
+ *
+ * @author Abhishek Tiwari
+ */
+@Alpha
+public class GobblinAWSClusterMaster extends GobblinClusterManager {
```

Comment:
Mostly because it is naturally a form of GobblinClusterManager, wherein it inherits most of the functionalities inbuilt in it and overrides a few methods. Using it as a member variable would need invoking methods on the member variable .. which otherwise are inherited